relays/ethereum-client/src/lib.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Tools to interact with (Open) Ethereum node using RPC methods.
+
+#![warn(missing_docs)]
+
+use crate::types::{
+	Address, Bytes, CallRequest, Header, HeaderWithTransactions, Receipt, SignedRawTx, Transaction, TransactionHash,
+	H256, U256,
+};
+
+use async_trait::async_trait;
+
+mod client;
+mod error;
+mod rpc;
+mod sign;
+
+pub use crate::client::Client as RpcClient;
+pub use crate::error::{Error, Result};
+pub use crate::sign::{sign_and_submit_transaction, SigningParams};
+
+pub mod types;
+
+/// Ethereum connection params.
+#[derive(Debug, Clone)]
+pub struct ConnectionParams {
+	/// Ethereum RPC host.
+	pub host: String,
+	/// Ethereum RPC port.
+	pub port: u16,
+}
+
+impl Default for ConnectionParams {
+	fn default() -> Self {
+		ConnectionParams {
+			host: "localhost".into(),
+			port: 8545,
+		}
+	}
+}
+
+/// The API for the supported Ethereum RPC methods.
+///
+/// Cloning client is a lightweight operation that only clones internal references.
+#[async_trait]
+pub trait Client: 'static + Send + Sync + Clone {
```

Comment:
I wonder if it makes sense to have this trait at all. Why not expose the `client::Client` directly with a bunch of `pub async fn` methods?
Do we have a different implementation for tests?

Reply:
The only reason I see for the trait is to move it to a yet another crate so that we can keep the dependency tree light. Now we import the `trait` but it always comes with a specific implementation, which kind of defeats the purpose of having the trait in the first place.

Reply:
I think initially maybe we were going to use some custom implementation in tests? But we now have higher-level traits that are actually customized for tests (`SourceClient`/`TargetClient`) => I agree that removing `Client` trait is a good deed. Removed